security-oauth2/src/test/groovy/io/micronaut/security/oauth2/client/DefaultOpenIdClientSpec.groovy

```diff
@@ -0,0 +1,81 @@
+package io.micronaut.security.oauth2.client
+
+import io.micronaut.context.ApplicationContext
+import io.micronaut.context.annotation.Requires
+import io.micronaut.core.util.CollectionUtils
+import io.micronaut.core.util.StringUtils
+import io.micronaut.http.HttpRequest
+import io.micronaut.http.annotation.Controller
+import io.micronaut.http.annotation.Get
+import io.micronaut.http.client.BlockingHttpClient
+import io.micronaut.http.client.HttpClient
+import io.micronaut.http.uri.UriBuilder
+import io.micronaut.runtime.server.EmbeddedServer
+import io.micronaut.security.annotation.Secured
+import io.micronaut.security.rules.SecurityRule
+import spock.lang.Issue
+import spock.lang.Specification
+
+class DefaultOpenIdClientSpec extends Specification {
+
+    @Issue("https://github.com/micronaut-projects/micronaut-security/issues/1199")
+    void "DefaultOpenIdClient is created correctly"() {
+        given:
+        EmbeddedServer google = ApplicationContext.run(EmbeddedServer, ["spec.name": "DefaultOpenIdClientSpecGoogle"])
```

Comment:
how beautiful is Micronaut Framework that you can create three servers in the same test seamlessly.